pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/store/TableViewLoadDataStoreImpl.java

```diff
@@ -65,48 +62,84 @@ public TableViewLoadDataStoreImpl(PulsarService pulsar, String topic, Class<T> c
             this.client = pulsar.getClient();
             this.topic = topic;
             this.clazz = clazz;
+            this.isShutdown = false;
         } catch (Exception e) {
             throw new LoadDataStoreException(e);
         }
     }
 
     @Override
     public synchronized CompletableFuture<Void> pushAsync(String key, T loadData) {
-        validateProducer();
-        return producer.newMessage().key(key).value(loadData).sendAsync().thenAccept(__ -> {});
+        String msg = validateProducer();
+        if (StringUtils.isNotBlank(msg)) {
+            return CompletableFuture.failedFuture(new IllegalStateException(msg));
+        }
+        return producer.newMessage().key(key).value(loadData).sendAsync()
+                .thenAccept(__ -> producerLastPublishTimestamp = System.currentTimeMillis());
     }
 
     @Override
     public synchronized CompletableFuture<Void> removeAsync(String key) {
-        validateProducer();
-        return producer.newMessage().key(key).value(null).sendAsync().thenAccept(__ -> {});
+        String msg = validateProducer();
+        if (StringUtils.isNotBlank(msg)) {
+            return CompletableFuture.failedFuture(new IllegalStateException(msg));
+        }
+        return producer.newMessage().key(key).value(null).sendAsync()
+                .thenAccept(__ -> producerLastPublishTimestamp = System.currentTimeMillis());
     }
 
     @Override
     public synchronized Optional<T> get(String key) {
-        validateTableView();
+        String msg = validateTableView();
+        if (StringUtils.isNotBlank(msg)) {
+            throw new IllegalStateException(msg);
+        }
         return Optional.ofNullable(tableView.get(key));
     }
 
     @Override
     public synchronized void forEach(BiConsumer<String, T> action) {
-        validateTableView();
+        String msg = validateTableView();
+        if (StringUtils.isNotBlank(msg)) {
+            throw new IllegalStateException(msg);
```

Comment:
Can we visibly mark this method will throw an exception, otherwise the caller may miss the catch block?

Reply:
I think this can be a runtime exception. I dont think we need a checked one here and make the caller explicitly catch this.